### PR TITLE
Removed dispaxis/dispdir

### PR DESCRIPTION
**Pull request by rcooke-ast**
Along with the new parser, I've removed all _horrid_ instances of dispaxis and dispdir throughout the cython and python code. A bug fix that came along with the new parser ensures that the dispaxis/dispdir is always 0 (i.e. the spectral direction is always along the zeroth axis). The development suite runs cleanly.

Algorithmically, the code is unchanged, just cleaned up.

### CODE REVIEW

**Comment by rcooke-ast** [comment]
A note to all - you will need to recompile the code with:

python setup.py develop

in the PYPIT folder for this change to be implemented. For the parser to work, you will need the future package, which can be installed with:

pip install future

File: pypit/arcyextract.pyx [hunk 1]
```diff
@@ -414,50 +413,50 @@ def extract_2d(np.ndarray[DTYPE_t, ndim=2] frame not None,
...
-                            print inlist
-                            print pixlarr
-                        uarea += area
-                        totflux += area*frame[j,pixcen[x]+i]
-                        toterror += cpow(area*error[j,pixcen[x]+i],2.0)
-                        if area == 0.0:
-                            if i>pixu+1:
+                    #area = poly_area(polyout)
+                    if area == np.nan:
+                        print "AREA=NAN (up)"
```
Comment (profxj) [inline pypit/arcyextract.pyx]: Please make these Python 3 compliant

File: pypit/arcyextract.pyx [hunk 2]
```diff
@@ -466,160 +465,42 @@ def extract_2d(np.ndarray[DTYPE_t, ndim=2] frame not None,
...
-                            print inlist
-                            print pixlarr
-                        darea += area
-                        totflux += area*frame[j,pixcen[x]+i]
-                        toterror += cpow(area*error[j,pixcen[x]+i],2.0)
-                        if area == 0.0:
-                            if -i>pixd+1:
+                    #area = poly_area(polyout)
+                    if area == np.nan:
+                        print "AREA=NAN (down)"
```
Comment (profxj) [inline pypit/arcyextract.pyx]: And these

File: pypit/arcyextract.pyx [hunk 3]
```diff
@@ -466,160 +465,42 @@ def extract_2d(np.ndarray[DTYPE_t, ndim=2] frame not None,
...
-                        if donesome == 1: break
-                    else:
-                        donesome = 1
-                    # If we haven't found overlap yet, then we must be off the chip
-                    if j-pixll>=5 and donesome==0:
-                        j = x+1
+                        if -i>pixd: pixd=-i
+                    i -= 1
+                # Break if we've already found some overlap
+                if uarea+darea == 0.0:
```
Comment (profxj) [inline pypit/arcyextract.pyx]: Maybe use  np.isclose()  ?

**Comment by profxj** [comment]
Have added a few inline comments.  Things look fine to me.

Can you add 'future' in the list of install requirements 
if you haven't already?

Merge when ready.

File: pypit/arcyextract.pyx [hunk 4]
```diff
@@ -730,38 +611,37 @@ def extract_mean(np.ndarray[DTYPE_t, ndim=2] frame not None,
...
-        if dispaxis == 0:
-            # yloc = slf._pixlocn[:,0,0]
-            # xloc = slf._pixlocn[0,:,1]
-            # ysiz = slf._pixlocn[:,0,2]
-            # xsiz = slf._pixlocn[0,:,3]
+        # yloc = slf._pixlocn[:,0,0]
+        # xloc = slf._pixlocn[0,:,1]
+        # ysiz = slf._pixlocn[:,0,2]
+        # xsiz = slf._pixlocn[0,:,3]
+        while True:
```
Comment (rcooke-ast) [inline pypit/arcyextract.pyx]: There shouldn't be. `j` is incremented by 1 on line 735, and there's a break statement once j reaches a certain value, on line 620.

**Comment by rcooke-ast** [comment]
The remaining comments are correct as they are in cython-speak. In fact, some of these functions are deprecated, and need to be removed completely. I'll do this after I've completed the echelle pipeline.

I had previously added the future requirement to the README (sorry, no PR for that one!).

**Pull request merged**